src/libraries/System.Private.CoreLib/src/System/SearchValues/Strings/Helpers/CharacterFrequencyHelper.cs

```diff
@@ -0,0 +1,127 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics;
+
+namespace System.Buffers
+{
+    internal static class CharacterFrequencyHelper
+    {
+        // Same as RegexPrefixAnalyzer.Frequency.
+        // https://github.com/dotnet/runtime/blob/a355d5f7db162714ee19533ca55074aa2cbd8a8c/src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexPrefixAnalyzer.cs#L956C43-L956C53
+        public static ReadOnlySpan<float> AsciiFrequency => new float[]
+        {
+            0.000f /* '\x00' */, 0.000f /* '\x01' */, 0.000f /* '\x02' */, 0.000f /* '\x03' */, 0.000f /* '\x04' */, 0.000f /* '\x05' */, 0.000f /* '\x06' */, 0.000f /* '\x07' */,
```

Comment:
I wonder whether we ought to have included a lot of ACGT for regex redux type inputs..

Reply:
With the current implementation, character frequency is only used as a heuristic for single-value sets and for the Aho-Corasick implementation.

The [interesting inputs](https://github.com/dotnet/performance/blob/af8bc7a227215626a51f427a839c55f90eec8876/src/benchmarks/micro/runtime/BenchmarksGame/regex-redux-5.cs#L61-L69) from Regex Redux will end up using the vectorized Teddy implementation where we do no such analysis currently.